File: symfony/src/Entity/Commentaire.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use App\Repository\CommentaireRepository;
use Symfony\Component\Validator\Constraints as Assert ;

/**
 * Commentaire
 *
 * @ORM\Table(name="commentaire", indexes={@ORM\Index(name="IDX_67F068BC4B89032C", columns={"post_id"})})
 * @ORM\Entity(repositoryClass=CommentaireRepository::class)
 */ 

 #[ORM\Table(name: "commentaire")]
 #[ORM\Entity(repositoryClass: CommentaireRepository::class)]
class Commentaire implements \JsonSerializable
{
    #[ORM\Column(name: "id", type: "integer", nullable: false)]
    #[ORM\Id]
    #[ORM\GeneratedValue(strategy: "IDENTITY")]
    private $id;

    #[ORM\ManyToOne]
#[ORM\JoinColumn(name: 'user_id', referencedColumnName: 'id_user')]
private ?User $user = null;

    #[ORM\Column(name: "imagec", type: "string", length: 255, nullable: true, options: ["default" => "NULL"])]
    private $imagec = 'NULL';

    #[ORM\Column(name: "descriptionc", type: "string", length: 255, nullable: false)]
    #[Assert\Length(min: 3)]
    #[Assert\NotNull]
    private $descriptionc;

    #[ORM\Column(name: "datec", type: "datetime", nullable: false, options: ["default" => "CURRENT_TIMESTAMP"])]
    private $datec = null;
    

    #[ORM\Column(name: "analyse_co", type: "string", length: 255, nullable: true, options: ["default" => "NULL"])]
    private $analyseCo = 'NULL';

    #[ORM\Column(name: "datec_ts", type: "datetime", nullable: true, options: ["default" => "CURRENT_TIMESTAMP"])]
    private $datecTs;

    #[ORM\ManyToOne(targetEntity: Post::class, inversedBy: "Commentaires")]
    #[ORM\JoinColumn(nullable: false)]
    private $post;

    // /**
    //  * @var \Post
    //  *
    //  * @ORM\ManyToOne(targetEntity="Post")
    //  * @ORM\JoinColumns({
    //  *   @ORM\JoinColumn(name="post_id", referencedColumnName="id")
    //  * })
    //  */
    // private $post;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getUser(): ?User
    {
        return $this->user;
    }

    public function setUser(?User $user): self
    {
        $this->user = $user;

        return $this;
    }

    public function getImagec(): ?string
    {
        return $this->imagec;
    }

    public function setImagec(?string $imagec): self
    {
        $this->imagec = $imagec;

        return $this;
    }

    public function getDescriptionc(): ?string
    {
        return $this->descriptionc;
    }

    public function setDescriptionc(string $descriptionc): self
    {
        $this->descriptionc = $descriptionc;

        return $this;
    }

    public function getDatec(): ?\DateTimeInterface
    {
        return $this->datec;
    }

    public function setDatec(\DateTimeInterface $datec): self
    {
        $this->datec = $datec;

        return $this;
    }


    public function getAnalyseCo(): ?string
    {
        return $this->analyseCo;
    }

    public function setAnalyseCo(?string $analyseCo): self
    {
        $this->analyseCo = $analyseCo;

        return $this;
    }

    public function getDatecTs(): ?\DateTimeInterface
    {
        return $this->datecTs;
    }

    public function setDatecTs(\DateTimeInterface $datecTs): self
    {
        $this->datecTs = $datecTs;

        return $this;
    }

    public function getPost(): ?Post
    {
        return $this->post;
    }

    public function setPost(?Post $post): self
    {
        $this->post = $post;

        return $this;
    }


     public function jsonSerialize(): array
     {
         return array(
             'id' => $this->id,
             'user' => $this->user,
             'post' => $this->post,
             'image' => $this->imagec,
             'description' => $this->descriptionc,
             'date' => $this->datec->format("d-m-Y"),
             'analyse' => $this->analyseCo,
             'datecTs' => $this->datecTs->format("d-m-Y")

         );
     }

     public function constructor($user, $post, $image, $description, $date, $analyse, $datecTs)
     {
         $this->user = $user;
         $this->post = $post;
         $this->imagec = $image;
         $this->descriptionc = $description;
         $this->datec = $date;
         $this->analyseCo = $analyse;
         $this->datecTs = $datecTs;
     }
}

File: mobile/src/com/fancytrade/entities/Commentaire.java
package com.fancytrade.entities;

import com.fancytrade.utils.DateUtils;
import com.fancytrade.utils.Statics;

import java.util.Date;

public class Commentaire implements Comparable<Commentaire> {

    private int id;
    private User user;
    private Post post;
    private String image;
    private String description;
    private Date date;
    private String analyse;
    private Date datecTs;

    public Commentaire() {
    }

    public Commentaire(int id, User user, Post post, String image, String description, Date date, String analyse, Date datecTs) {
        this.id = id;
        this.user = user;
        this.post = post;
        this.image = image;
        this.description = description;
        this.date = date;
        this.analyse = analyse;
        this.datecTs = datecTs;
    }

    public Commentaire(User user, Post post, String image, String description, Date date, String analyse, Date datecTs) {
        this.user = user;
        this.post = post;
        this.image = image;
        this.description = description;
        this.date = date;
        this.analyse = analyse;
        this.datecTs = datecTs;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Post getPost() {
        return post;
    }

    public void setPost(Post post) {
        this.post = post;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public String getAnalyse() {
        return analyse;
    }

    public void setAnalyse(String analyse) {
        this.analyse = analyse;
    }

    public Date getDatecTs() {
        return datecTs;
    }

    public void setDatecTs(Date datecTs) {
        this.datecTs = datecTs;
    }


    @Override
    public int compareTo(Commentaire commentaire) {
        switch (Statics.compareVar) {
            case "User":
                return this.getUser().getEmail().compareTo(commentaire.getUser().getEmail());
            case "Post":
                return this.getPost().getSujet().compareTo(commentaire.getPost().getSujet());
            case "Image":
                return this.getImage().compareTo(commentaire.getImage());
            case "Description":
                return this.getDescription().compareTo(commentaire.getDescription());
            case "Date":
                DateUtils.compareDates(this.getDate(), commentaire.getDate());
            case "Analyse":
                return this.getAnalyse().compareTo(commentaire.getAnalyse());
            case "DatecTs":
                DateUtils.compareDates(this.getDatecTs(), commentaire.getDatecTs());

            default:
                return 0;
        }
    }

}